### PR TITLE
MB-5054 - Port run_ENV_migrations make commands from com to gov

### PR DESCRIPTION
**Pull request by dynamike**
## Description

Right now the run_env_migrations makefile endpoints are still tied to the legacy commercial environment. This PR updates things to use GovCloud. In this process it also decouples the logic to call out to the vault go library as that does not support assuming IAM roles between AWS accounts.

## Reviewer Notes

I'll be updating the https://github.com/transcom/mymove/wiki/migrate-the-database wiki as part of this PR

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make run_exp_migrations 
make run_stg_migrations 
make run_prd_migrations
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

### CODE REVIEW

**Review by travelar — APPROVED** [review]

**Review by kilbergr — CHANGES_REQUESTED** [review]
Please update references to `run_ENV_migrations` in documentation to reflect this change.

**Review by gidjin — COMMENTED** [review]
I'm in the process of running the migrations. One thing is that this setup requires a long command line, which we are not used to on milmove. I'm wondering if we can put `DISABLE_AWS_VAULT_WRAPPER=1 AWS_REGION=us-gov-west-1 aws-vault exec transcom-gov-milmove-exp` inside of the make command `make run_exp_migrations` to make the transition simpler.

**Comment by kilbergr** [comment]
I would like to see an app eng run it. I'm wondering whether we can keep the make command a little simpler by putting the DISABLE part in the command like on line 399-402.

**Comment by kilbergr** [comment]
Oh I see @gidjin s response now--yeah I am on a similar page.

**Comment by dynamike** [comment]
@gidjin @kilbergr I simplified things a bit. Can y'all test it out?

**Review by gidjin — CHANGES_REQUESTED** [review]
The stg and exp migrations ran successfully! 🎉  However, the prd migrations failed with the below error, to make sure it wasn't a fluke I ran it twice. Marking this as changes requested due to that. Would also like to see the commands simplified if possible.

```sh
[POP] 2020/11/12 16:39:38 info - > add_april_fuel_eia_diesel_prices
[POP] 2020/11/12 16:39:38 info - > update_superusers
[POP] 2020/11/12 16:39:38 info - > add_new_tdls
AWS Error Code: "AccessDenied", "Access Denied", %!q(<nil>)
[POP] 2020/11/12 16:39:38 info - 9.7493 minutes
panic: error running migrations: error reading migration file at "s3://transcom-gov-milmove-prd-app-us-gov-west-1/secure-migrations/20190410215335_add_05_2019_tspps.up.sql": AccessDenied: Access Denied
        status code: 403, request id: D619D35998EA906D, host id: 0LRQSu4IGLXHzsjuem2pIfXcAx4NbvNabVJgE7YgPJYvS7bjopvI9F2C3OPsiWGjW18ktoxiigk=

goroutine 1 [running]:
main.main()
        /Users/john/projects/dod/mymove/cmd/milmove/main.go:122 +0x159a
make: *** [run_prd_migrations] Error 2
```

**Comment by dynamike** [comment]
The looks like a IAM perms issue on the AWS side of things

**Comment by gidjin** [comment]
Update short commands are working great! Thanks for making that change. Let me know when to test the prd migrations again.

**Review by gidjin — APPROVED** [review]
Changes work for me! :shipit: Reminder to modify the docs https://github.com/transcom/mymove/wiki/migrate-the-database once it's merged or shortly before and to make an announcement in #prac-engineering and maybe even the next demo

**Review by kilbergr — CHANGES_REQUESTED** [review]
Please update references to run_ENV_migrations in documentation to reflect this change.

**Comment by dynamike** [comment]
@kilberg I've update the docs.

**Review by kilbergr — CHANGES_REQUESTED** [review]
I'm referring to references within MyMove: https://github.com/transcom/mymove/blob/master/scripts/pricing-import#L126 and https://github.com/transcom/mymove/blob/master/scripts/generate-secure-migration#L77. Since you've changed the names of the make commands, you have to update these as well.

File: scripts/pricing-import [hunk 1]
```diff
@@ -123,7 +123,7 @@ Next:
     2. If everything looks good, upload the migration to S3 with this utility:
        scripts/upload-secure-migration \\
           ${prod_migration_name}
-    3. Run make run_prod_migrations to verify that the upload worked and that the migration can be applied successfully. This will check against the commercial environment at the moment.
+    3. Run make run_prd_migrations to verify that the upload worked and that the migration can be applied successfully. This will check against the commercial environment at the moment.
```
Comment (kilbergr) [inline scripts/pricing-import:126]:
```suggestion
    3. Run make run_prd_migrations to verify that the upload worked and that the migration can be applied successfully.
```

File: scripts/generate-secure-migration [hunk 2]
```diff
@@ -74,7 +74,7 @@ Next:
     3. If everything looks good, upload the migration to S3 with this utility:
        scripts/upload-secure-migration \\
           ${prod_migration_name}
-    4. Run make run_prod_migrations to verify that the upload worked and that the migration can be applied successfully. This will check against the commercial environment only at the moment.
+    4. Run make run_pr_migrations to verify that the upload worked and that the migration can be applied successfully. This will check against the commercial environment only at the moment.
```
Comment (kilbergr) [inline scripts/generate-secure-migration:77]:
```suggestion
    4. Run make run_prd_migrations to verify that the upload worked and that the migration can be applied successfully.
```

**Review by kilbergr — CHANGES_REQUESTED** [review]
Caught a typo and removed some no longer correct info. Once this is changed, will approve.

**Review by kilbergr — APPROVED** [review]

**Pull request merged**